### ISSUE TITLE
All metadata categories that are not inferred cell types show only one value ("Not defined") in the plots

### ISSUE THREAD

**Issue by pcm32** [ebi-gene-expression-group/atlas-web-single-cell#180]
For every metadata field that I’m choosing that is not inferred cell type, I get only “Not defined” values. See for instance https://wwwdev.ebi.ac.uk/gxa/sc/experiments/E-MTAB-9067/results?colourBy=age or any other metadata field there and compare to https://www.ebi.ac.uk/gxa/sc/experiments/E-MTAB-9067/results?colourBy=metadata&metadata=age

In public/fallback:

![image](https://user-images.githubusercontent.com/368478/126835027-757e808f-ae32-48d2-8a18-48751f5d857d.png)

Whereas in dev:

![image](https://user-images.githubusercontent.com/368478/126835065-c59e5f48-7440-489b-b94c-b188999bd68e.png)

I see this behaviour across all the experiments I have tried so far. It doesn't seem to happen though with "Inferred cell type fields".

**Comment by pcm32** [comment]
The data does make it to the tooltip though, so I suspect this is just an issue with the plotting. I had also checked before, and the data is there in solr.

![image](https://user-images.githubusercontent.com/368478/126835530-be2c03ce-5546-45e9-8414-29c2ca095469.png)

**Comment by pcm32** [comment]
After conversation with @upendrakumbham it seems to be that the current codebase is expecting to retrieve the metadata from the database, but like the tooltip, this needs to come from the the scxa-analytics Solr index where cells metadata is stored (and that should include the inferred celltypes). Currently inferred cell types coloring works because we have groups of cells in the database for inferred cell types (this is needed for generating marker genes for the anatomograms and heatmap by cell types), but this shouldn't be the source for metadata for that category either. Probably the part of the code that was doing that got washed out when bringing in the UMAP changes.

**Comment by upendrakumbham** [comment]
after discussion with @pcm32, I verified the code again to see any contacts with the Solr while dealing with cluster and expression end points.  

This is my view after the code walkthrough:
When I compare with previous release endpoints, these endpoints got changed as per UMAP feature:
- `/json/experiments/{experimentAccession}/tsneplot/{parameter}/clusters/variable/{variable}`

Corresponding SQL query: 

- `/json/experiments/{experimentAccession}/tsneplot/{plotOption}/expression`

These two endpoints are contacting database to get data like `x,y,cell_id,cluster_id`,expression_level. During this process, I didn't see any contact to Solr to fetch metadata. This behavior is the same for the previous release also (202104).

Once we received a set of `cell_id`s. For the Tooltip, we are calling another metadata endpoint: 
- json/experiment/{experimentAccession}/cell/{cellId}/metadata

This endpoint contacting Solr and getting metadata for each cell_id. we repeat this endpoint calling for each cell-id to see corresponding metadata.
